composer.json

```diff
@@ -15,7 +15,7 @@
   "require": {
     "php": "^7.2 || ^8.0",
     "ext-simplexml": "*",
-    "phpstan/phpstan": "^1.10.36"
+    "phpstan/phpstan": "^1.10.49"
```

Comment:
since the behaviour we fix with this PR only affects 1.10.49+
refs https://github.com/phpstan/phpstan-src/pull/2818

Reply:
I disagree with this change. The dynamic return type extension is valid and works for older PHPStan versions too.